### PR TITLE
Enzo frontend

### PR DESCRIPTION
**Pull request by milealeonard**

### CODE REVIEW

File: datamarket/components/MyDatasets/ListDatasets.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import {SideBar} from "./SideBar";
+
+
+export const ListDatasets = ({
+  forOwnersOnly,
+  datasets,
+  accounts,
+  liteMode,
+}: {
+  forOwnersOnly?: boolean;
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/ListDatasets.tsx]: smal styling thing: put optional params at the end

File: datamarket/components/MyDatasets/ListDatasets.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import AddIcon from '@mui/icons-material/Add';
+import { DatasetView } from "./DatasetView";
+import { useRouter } from "next/router";
+import SettingsIcon from '@mui/icons-material/Settings';
+import styles from './ListDatasets.module.css';
+import ClearIcon from '@mui/icons-material/Clear';
+import {SideBar} from "./SideBar";
+
+
+export const ListDatasets = ({
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/ListDatasets.tsx:19]: adding 
```
/**
* some info about what the different params are
**/
```
is useful to define params when theyre not super obvious
ie why is liteMode useful.

File: datamarket/components/MyDatasets/ListDatasets.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  liteMode,
+}: {
+  forOwnersOnly?: boolean;
+  datasets: Dataset[];
+  accounts: string[];
+  liteMode?: boolean;
+}): React.ReactElement => {
+  const router = useRouter();
+  const [sidePanelHidden, setSidePanelHidden] = React.useState(true);
+  const [sortedDataSet, setDatasets] = React.useState(datasets);
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/ListDatasets.tsx]: name on left and name on right should be

[someName, setSomeName]

File: datamarket/components/MyDatasets/AddButton.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          >
+            <AddIcon sx={{ width: "64px", height: "64px" }} />
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className={styles.add}>
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/AddButton.tsx]: i would encourage you to go back and fix the stylings to all be tailwind.
just cleaner to only stick to one method
can ask chat gpt for help + if you remember you already added a custom tailwind css thing.

File: datamarket/components/MyDatasets/DatasetActions.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  publicIndex,
+  dataset,
+  accounts,
+  showIcons,
+}: {
+    isOwner?: boolean;
+    publicIndex: number | undefined;
+    dataset: DatasetWithIndex;
+    accounts: string[];
+    showIcons: boolean;
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/DatasetActions.tsx]: make sure to format your documents. this seems out of wack

File: datamarket/components/MyDatasets/DatasetActions.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  const navToUpdate = (): void => {
+    router.push(`update/${publicIndex}`);
+  };
+
+  const downloadDataset = (): void => {
+    decryptAndDownload({ cid: dataset.data, account: accounts[0] });
+  };
+  return (
+    <div className="">
+      {showIcons && (
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/DatasetActions.tsx]: if we dont want to render anything when showIcons is false, then it should be
```
if (!showIcons) {
return <></>
}
return (
the actual thing
)
```

File: datamarket/components/MyDatasets/DatasetView.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  viewOption,
+}: DatasetViewProps): React.ReactElement => {
+  const router = useRouter();
+  const isOwner = userOwnsDataset(dataset, accounts);
+  const shouldShowThisDataset = !forOwnersOnly || isOwner;
+  const { width } = useWindowSize();
+
+  const purchaseDatasetWrapper = async (): Promise<void> => {
+    if (publicIndex === undefined) {
+      console.error("AHHHHH");
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/DatasetView.tsx]: lol could make this more descriptive

File: datamarket/constants.ts [hunk 8]
```diff
@@ -2,6 +2,12 @@
 // INTERFACES/TYPES //
 //////////////////////
 
+import internal from "stream";
+
+export type SortType = "Price ($-$$$)" | "Price ($$$-$)" | "Name (A-Z)" | "Name (Z-A)";
+export type ViewType = "Gallery" | "List";
```
Comment (milealeonard) [inline datamarket/constants.ts]: this is fine, but an enum is usually better.
ie
`export enum ViewType {GALLERY = "Gallery"}` (or something, u can ask chatgpt)
then you use it like
ViewType.GALLERY

File: datamarket/components/MyDatasets/ListDatasets.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    setSidePanelHidden(!sidePanelHidden);
+  };
+
+  React.useEffect(() => {
+    console.log(allDatasets);
+    console.log(filter);
+    setDatasets(() => {
+      return sortDatasets(filterDatasets(allDatasets, filter), sortOption);
+    });
+  }, [sortOption, filter]);
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/ListDatasets.tsx]: i dont like this tbh
should try to remove this and instead have this be called by an event handler
ie whenever a user clicks somethign which changes the sortOption or filter, itll call "reSortDatasets" which will perform this action.
Lmk if thats unclear

File: datamarket/components/MyDatasets/ListDatasets.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+  const toggleSidePanel = (): void => {
+    setSidePanelHidden(!sidePanelHidden);
+  };
+
+  React.useEffect(() => {
+    console.log(allDatasets);
+    console.log(filter);
+    setDatasets(() => {
+      return sortDatasets(filterDatasets(allDatasets, filter), sortOption);
+    });
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/ListDatasets.tsx]: this seems like a bug. we're setting datasets to a function?

File: datamarket/components/MyDatasets/ListDatasets.tsx [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+  }, [sortOption, filter]);
+
+  return (
+    <div className="flex flex-col w-full">
+      <AddButton />
+      <div className={`${styles.galleryContainer}`}>
+        {datasets.map((dataset: DatasetWithIndex) => {
```
Comment (milealeonard) [inline datamarket/components/MyDatasets/ListDatasets.tsx]: we're not using `allDatasets`? no point to the useeffect above i think

File: datamarket/components/MyDatasets/DatasetActions.tsx (hunk 5 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/DatasetActions.tsx]: Done

File: datamarket/components/MyDatasets/ListDatasets.tsx (hunk 2 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/ListDatasets.tsx:19]: Done

File: datamarket/components/MyDatasets/ListDatasets.tsx (hunk 1 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/ListDatasets.tsx]: Done

File: datamarket/constants.ts (hunk 8 above)
Comment (EnzoMassyle) [reply datamarket/constants.ts]: Done

File: datamarket/components/MyDatasets/AddButton.tsx (hunk 4 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/AddButton.tsx]: Done

File: datamarket/components/MyDatasets/DatasetActions.tsx (hunk 6 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/DatasetActions.tsx]: Done

File: datamarket/components/MyDatasets/DatasetView.tsx (hunk 7 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/DatasetView.tsx]: Done

File: datamarket/components/MyDatasets/ListDatasets.tsx (hunk 11 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/ListDatasets.tsx]: Done

File: datamarket/components/MyDatasets/ListDatasets.tsx (hunk 10 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/ListDatasets.tsx]: Done. Seemed to me like this was okay but I was able to get the behavior needed w/o a useEffect, just a simple callback

File: datamarket/components/MyDatasets/ListDatasets.tsx (hunk 9 above)
Comment (EnzoMassyle) [reply datamarket/components/MyDatasets/ListDatasets.tsx]: Done

**Pull request merged**